pkg/app/nginx.go

```diff
@@ -0,0 +1,62 @@
+package app
+
+import (
+	"github.com/maistra/maistra-test-tool/pkg/examples"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+)
+
+type nginx struct {
+	ns   string
+	mTLS bool
+}
+
+var _ App = &nginx{}
+
+func Nginx(ns string) App {
+	return &nginx{ns: ns, mTLS: false}
+}
+
+func NginxWithMTLS(ns string) App {
+	return &nginx{ns: ns, mTLS: true}
+}
+
+func (a *nginx) Name() string {
+	return "nginx"
+}
+
+func (a *nginx) Namespace() string {
+	return a.ns
+}
+
+func (a *nginx) Install(t test.TestHelper) {
```

Comment:
T10, T14 and T15 will use this app/nginx.go for nginx installation. @pbajjuri20 , we may watch this change here and then use the app.InstallAndWaitReady method in our PR later.